RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsProducerPortable.cc

```diff
@@ -0,0 +1,141 @@
+        weightsCache_(PortableHostObject<EcalUncalibRecHitPhase2Weights>(
+            cms::alpakatools::host(), [](const edm::ParameterSet &ps) {
+              EcalUncalibRecHitPhase2Weights weights;
+              const auto amp_weights = ps.getParameter<std::vector<double>>("weights");
+              const auto timeWeights = ps.getParameter<std::vector<double>>("timeWeights");
+              for (unsigned int i = 0; i < ecalPh2::sampleSize; ++i) {
+                if (i < amp_weights.size()) {
+                  weights.weights[i] = static_cast<float>(amp_weights[i]);
+                } else {
+                  weights.weights[i] = 0;
+                }
+                if (i < timeWeights.size()) {
+                  weights.timeWeights[i] = static_cast<float>(timeWeights[i]);
+                } else {
+                  weights.timeWeights[i] = 0;
+                }
+              }
+              return weights;
+            }(ps))) {}
+
+  void EcalUncalibRecHitPhase2WeightsProducerPortable::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<std::string>("uncalibratedRecHitsLabelEB", "EcalUncalibRecHitsEB");
+    //The below weights values should be kept up to date with those on the CPU version of this module
+    desc.add<std::vector<double>>("weights",
+                                  {-0.121016,
+                                   -0.119899,
+                                   -0.120923,
+                                   -0.0848959,
+                                   0.261041,
+                                   0.509881,
+                                   0.373591,
+                                   0.134899,
+                                   -0.0233605,
+                                   -0.0913195,
+                                   -0.112452,
+                                   -0.118596,
+                                   -0.121737,
+                                   -0.121737,
+                                   -0.121737,
+                                   -0.121737});
+    desc.add<std::vector<double>>("timeWeights",
+                                  {0.429452,
+                                   0.442762,
+                                   0.413327,
+                                   0.858327,
+                                   4.42324,
+                                   2.04369,
+                                   -3.42426,
+                                   -4.16258,
+                                   -2.36061,
+                                   -0.725371,
+                                   0.0727267,
+                                   0.326005,
+                                   0.402035,
+                                   0.404287,
+                                   0.434207,
+                                   0.422775});
+
+    desc.add<edm::InputTag>("digisLabelEB", edm::InputTag("simEcalUnsuppressedDigis", ""));
```

Comment:
Are the `simEcalUnsuppressedDigis` in a portable collection in SoA format ?
Otherwise a better default could be
```suggestion
    desc.add<edm::InputTag>("digisLabelEB", edm::InputTag("ecalPhase2DigiToPortableProducer", "ebDigis"));
```